packages/flutter/lib/src/widgets/scrollable.dart

```diff
@@ -501,21 +501,24 @@ class ScrollableState extends State<Scrollable> with TickerProviderStateMixin
   @override
   Widget build(BuildContext context) {
     assert(position != null);
-    // TODO(ianh): Having all these global keys is sad.
-    Widget result = RawGestureDetector(
-      key: _gestureDetectorKey,
-      gestures: _gestureRecognizers,
-      behavior: HitTestBehavior.opaque,
-      excludeFromSemantics: widget.excludeFromSemantics,
-      child: Semantics(
-        explicitChildNodes: !widget.excludeFromSemantics,
-        child: IgnorePointer(
-          key: _ignorePointerKey,
-          ignoring: _shouldIgnorePointer,
-          ignoringSemantics: false,
-          child: _ScrollableScope(
-            scrollable: this,
-            position: position,
+    /// [_ScrollableScope] is voluntarily placed above [RawGestureDetector]
+    /// so that it is possible to obtain the [ScrollableState]
+    /// from [ScrollNotifications] using [Scrollable.of]
```

Comment:
this comment doesn't really explain what's going on to me. For example, it doesn't mention that `_gestureDetectorKey` is used for sending the notifications, and that the `context` passed to `Scrollable.of` needs to be able to see the `_ScrollableScope`, and that therefore the notifications context needs to be one that's a descendant of `_ScrollableScope`, which thus means the `RawGestureDetector` must be below the `_ScrollableScope`.

Reply:
Done. I added a mention to `notificationContext`, as it explains the relationship with `_gestureDetectorKey`